### PR TITLE
Alteration Curse + Amulet of Digital Invisibility

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds new item, medal of alteration for heretics.
Replaces the paralysis curse with alteration curse.

Also makes digital camoflage a trait now. Before it used to be a boolean instead of mob class.

## Why It's Good For The Game

Makes curse of paralysis a little bit better.
The digital camoflage replaces the useless corrosion curse. May be worth investing a point into it.

Digital camo trait because spaghetti code.

## Changelog
:cl:
add: Alteration curse replace paralysis curse, available for Ash and Flesh.
add: Amulet of Digital Invisibility replaces corrosion curse, available to Ash and Rust, craftable from a shard and circuitboard.
del: Paralysis curse and Corrosion curse are now removed.
tweak: Digital camoflage is now a trait.
/:cl:

### CODE REVIEW

**Comment by eeSPee** [comment]
- [ ] Amulet needs sprite

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 1]
```diff
@@ -342,49 +351,102 @@
...
 //	---	CURSES ---
 
-/datum/eldritch_knowledge/curse/corrosion
-	name = "Curse of Corrosion"
-	gain_text = "Cursed land, cursed man, cursed mind."
-	desc = "Curse someone for 2 minutes of vomiting and major organ damage. Using a wirecutter, a heart, and an item that the victim touched  with their bare hands."
+/datum/eldritch_knowledge/curse/alteration
+	name = "Curse Of Alteration"
+	gain_text = "Mortal bodies, prisons of flesh. Death, a release..."
+	desc = "Start an alteration ritual by transmuting a wire cutter a hatchet and an item that the victim touched with their bare hands. Inflict a debilitating curse that will cripple your target's body for 2 minutes. Add eyes, ears, limbs or tongues to the mix to disable those organs while the curse is in effect."
```
Comment (Sarchutar) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]:
```suggestion
	desc = "Start an alteration ritual by transmuting a wire cutter, a hatchet and an item that the victim touched with their bare hands. Inflict a debilitating curse that will cripple your target's body for 2 minutes. Add eyes, ears, limbs or tongues to the mix to disable those organs while the curse is in effect."
```

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 2]
```diff
@@ -154,6 +154,12 @@
 	desc = "A strange medallion. Peering through the crystalline surface, the light refracts into new and terrifying spectrums of color. You see yourself, reflected off cascading mirrors, warped into impossible shapes."
 	trait = TRAIT_XRAY_VISION
 
+/obj/item/clothing/neck/eldritch_amulet/guise
+	name = "Guise of Istasha"
```
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm]:
```suggestion
	name = "guise of Istasha"
```

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 3]
```diff
@@ -342,49 +351,102 @@
...
+			debuffs |= "tongue"
+		else if(istype(atom_in_range,/obj/item/organ/eyes))
+			extra_atoms |= A
+			debuffs |= "eyes"
+		else if(istype(atom_in_range,/obj/item/organ/ears))
+			extra_atoms |= A
+			debuffs |= "ears"
+		else if(istype(atom_in_range,/obj/item/organ/liver) || istype(atom_in_range,/obj/item/organ/lungs) || istype(atom_in_range,/obj/item/organ/appendix) || istype(atom_in_range,/obj/item/organ/heart))
+			extra_atoms |= A
+			debuffs |= "organs"
```
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: Use a switch statement here
`switch(atom_in_range.type)`

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 4]
```diff
@@ -342,49 +351,102 @@
...
-/datum/eldritch_knowledge/curse/paralysis/curse(mob/living/chosen_mob)
-	. = ..()
-	ADD_TRAIT(chosen_mob,TRAIT_PARALYSIS_L_LEG,MAGIC_TRAIT)
-	ADD_TRAIT(chosen_mob,TRAIT_PARALYSIS_R_LEG,MAGIC_TRAIT)
-	chosen_mob.update_mobility()
+	if (chosen_mob.has_status_effect(/datum/status_effect/corrosion_curse))
+		return FALSE
 
-/datum/eldritch_knowledge/curse/paralysis/uncurse(mob/living/chosen_mob)
+	var/mob/living/carbon/human/chosen_mortal = chosen_mob
```
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:413]: target might not be human

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm (hunk 3 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: Will that check for subtypes?

Comment (utilbird) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: oh you're right, mb. There has to be a better way to do this though. An associated list with the organ debuff comes into mind.

**Review by utilbird — CHANGES_REQUESTED** [review]
Yanderedev would be proud 🥛

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm (hunk 3 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: Ok. I've done it a little bit different. It checks organ/bodypart type and adapts accordingly.

**Review by utilbird — APPROVED** [review]
Nice job 👍

**Pull request merged**